ui/packages/catalog-ui-search/src/main/webapp/behaviors/dropdown.behavior.js

```diff
@@ -0,0 +1,255 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/*global require*/
+const Behaviors = require('./Behaviors');
+const Marionette = require('marionette');
+const _ = require('lodash');
+const $ = require('jquery');
+const CustomElements = require('js/CustomElements');
+const DropdownBehaviorUtility = require('./dropdown.behavior.utility');
+
+const tagName = CustomElements.register('behavior-dropdown');
+Behaviors.addBehavior('dropdown', Marionette.Behavior.extend({
+    events() {
+        return this.options.dropdowns.reduce((eventMap, dropdown) => {
+            eventMap[`click ${dropdown.selector}`] = this.generateHandleClick(dropdown);
+            eventMap[`keydown ${dropdown.selector}`] = this.generateHandleEnter(dropdown);
+            eventMap[`keyup ${dropdown.selector}`] = this.generateHandleSpace(dropdown);
+            return eventMap;
+        }, {}); 
+    },
+    // We can't do this on initialize because view.cid isn't set yet.  OnFirstRender is closest to what we want.
+    onFirstRender() {
+        this.listenForOutsideInteraction();
+        this.listenForResize();
+    },
+    /*
+        buttons don't take action until keyup for space in browsers, try it for yourself
+        https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_button_test
```

Comment:
❓Is this beneficial to have this in with the link?

Reply:
I like it 🤷‍♂️